pkg/apis/provisioning/v1alpha5/requirements.go

```diff
@@ -298,7 +298,9 @@ func (r *Requirements) MarshalJSON() ([]byte, error) {
 
 func (r *Requirements) UnmarshalJSON(b []byte) error {
 	var requirements []v1.NodeSelectorRequirement
-	json.Unmarshal(b, &requirements)
+	if err := json.Unmarshal(b, &requirements); err != nil {
+		return err
```

Comment:
nit: could you change the erroring to fit how we do it elsewhere in the repo?
```suggestion
		return fmt.Errorf("unmarshalling json, %w", err)
```

Reply:
I'd actually argue against this, since it's a wrapper function. We want our errors to model the business logic flow, but not necessarily the exact implementation. We have two levels of unmarshalling json, so we probably want to return the naked error to avoid repeating ourselves.

Reply:
I see both styles in the repo I agree with @ellistarn on this.